src/pipert2/core/handlers/message_handlers/queue_handler.py

```diff
@@ -58,3 +57,7 @@ def _put(self, message: bytes):
             self.output_queue.put(message, block=self.blocking, timeout=self.timeout)
         except Full:
             self.logger.exception("The queue is full!")
+
+    def teardown(self):
+        if self.input_queue.mp_queue is not None:
+            self.input_queue.mp_queue.put(None)
```

Comment:
Maybe have these lines in the queue  wrapper class and call it from there ?
Call the function like stop input_queue_execution or something like that.